client/http/client.go

```diff
@@ -342,6 +367,13 @@ func (c *client) WithBackoffer(bo *retry.Backoffer) Client {
 	return &newClient
 }
 
+// WithTimeout sets and returns a new client with a new `http.Client` whose timeout is the given one.
+func (c *client) WithTimeout(dur time.Duration) Client {
+	newClient := *c
+	newClient.inner = c.inner.clone(withInnerTimeout(dur))
+	return &newClient
+}
```

Comment:
The initial purpose of introducing `clientInner` is to allow safe sharing of the same inner core, even with multiple `Client` interfaces. This ensures that common components such as service discovery are reused effectively. However, this constraint appears to be violated. What about just keeping the timeout option as an initialization parameter instead of a dynamic variable?

Reply:
Or we could make the timeout an upper layer's behavior rather than the HTTP client configuration.

Reply:
Currently, Common components don't rely on `clientInner(http.Client)`. So I think it's safe to clone the `clientInner`.  However, I do not consider this parameter to be a dynamic variable because it creates a new `clientInner` without affecting the previous `clientInner`.

In fact, the reason of why I do this is because the timeout of `http. Client` in the store helper in TiDB is 0.  However, the timeout value of http. Client in client-go is 30 seconds.  Setting a default timeout for `http.Client` eliminates the need to set a timeout in context each time for most requests.